pkg/policy/api/utils.go

```diff
@@ -44,6 +45,25 @@ func (h *PortRuleHTTP) Equal(o PortRuleHTTP) bool {
 			return false
 		}
 	}
+
+	for i, value := range h.HeaderMatches {
+		if !o.HeaderMatches[i].Equal(value) {
+			return false
+		}
+	}
+	return true
+}
+
+// Equal returns true if both HTTP rules are equal
+func (h *HeaderMatch) Equal(o *HeaderMatch) bool {
+	if h.Mismatch != o.Mismatch ||
+		h.Name != o.Name ||
+		h.Value != o.Value ||
+		h.Secret != nil && o.Secret == nil ||
+		h.Secret == nil && o.Secret != nil ||
+		h.Secret != nil && o.Secret != nil && *h.Secret != *o.Secret {
```

Comment:
Can this use `!h.Secret.Equal(o.Secret)`? (Ignore if this is autogenerated..)

Reply:
`Secret.Equal()` did not exist, but I factored it out of this for clarity.